packages/manager/modules/pci/src/projects/project/storages/databases/components/tags-input/tags-input.controller.js

```diff
@@ -0,0 +1,37 @@
+import remove from 'lodash/remove';
+import pull from 'lodash/pull';
+
+export default class PciTagsInputController {
+  $onInit() {
+    this.items = [];
+    this.tag = '';
+    this.model.forEach((element) => {
+      this.items.push({ title: element });
+    });
+  }
+
+  addTag() {
+    if (!this.isDisabled()) {
+      this.model.push(this.tag);
+      this.items.push({ title: this.tag });
+      this.tag = '';
+    }
+  }
+
+  removeTag(tag) {
+    pull(this.model, tag.title);
+    remove(this.items, (i) => i.title === tag);
+  }
+
+  isDisabled() {
+    const isEmptyOrSpaces =
+      this.tag === null || this.tag.match(/^ *$/) !== null;
+    return isEmptyOrSpaces || this.model.indexOf(this.tag) !== -1;
+  }
+
+  handleKeyPressed(key) {
+    if (key.which === 13) {
```

Comment:
this will not work with devices which does not have keyboard like mobile and ipod.

Reply:
Most mobile browsers fire the key event with the 13 code when the enter button is pressed on the virtual keyboard.  By the way, there also is a button that does the same thing, so the user will still be able to add a tag.